bin/main.ml

```diff
@@ -1,18 +1,60 @@
 open Diffcessible
 open Cmdliner
+open ExtUnix.Specific
+
+let create_empty_diff_content () =
+  "diff --git a/Error b/Error\n--- a/Error\n+++ b/Error\n"
+
+let with_input_fd file_path f =
+  match file_path with
+  | Some path ->
```

Comment:
No need to check whether `file_path` is `Some` or `None` here, this function is only ever called with `Some path`. It can be simplified.